docs/source/contributing.rst

```diff
@@ -530,9 +530,10 @@ Track Attributes
 Custom track attributes should be global, track-level data.
 For some datasets, there is a separate, dataset-level metadata file
 with track-level metadata, e.g. as a csv. When a single file is needed
-for more than one track, we recommend using writing a ``_load_metadata`` method
-and passing it to a ``LargeData`` object, which is available globally throughout 
-the module to avoid loading the same file multiple times.
+for more than one track, we recommend using writing a ``_metadata`` cached property
+in the Dataset class (see the dataset module example code above). When this is specified,
+it will populate a tracks hidden `_track_metadata` field, which can be accessed from
```

Comment:
I think we should make explicit that `metadata` has to be an index-like dictionary when track-level information is available, maybe add a small `metadata` section explaining the cases or something like that? I'm afraid that because `_track_metadata` is hidden it might be a bit obscure for the contributor if we don't explain more.